### PR TITLE
[prometheus-json-exporter]: Allow additional container args and service monitor relabelings

### PR DESCRIPTION
**Pull request by faisal-khalique**
#### What this PR does / why we need it
- Adds the option to pass arguments to the container. Could be needed for many reason. For example collecting logstash metrics with many pipelines where the pipeline definitions vary based on environment. Having the option of providing argument `--config.file` will greatly simplify such deployments. Considering future use cases, adding an array instead of just specifying the argument `--config.file` now
- Add additional relabelings for service-monitor which can be used to drop unwanted labels from the metrics like `container`, `pod`, etc.

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-json-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -42,6 +42,13 @@ spec:
       {{- end }}
       containers:
         - name: {{ .Chart.Name }}
+          {{- if .Values.extraArgs }}
```
Comment (zanhsieh) [inline charts/prometheus-json-exporter/templates/deployment.yaml:45]:
```suggestion
          {{- with .Values.extraArgs }}
````

File: charts/prometheus-json-exporter/templates/deployment.yaml [hunk 2]
```diff
@@ -42,6 +42,13 @@ spec:
...
+          args:
+          {{- range .Values.extraArgs }}
+          - {{ .name | quote }}
+          - {{ .value | quote }}
```
Comment (zanhsieh) [inline charts/prometheus-json-exporter/templates/deployment.yaml:49]: I don't think this correct.

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/prometheus-json-exporter/templates/deployment.yaml (hunk 2 above)
Comment (faisal-khalique) [reply charts/prometheus-json-exporter/templates/deployment.yaml:49]: I believe it is. The changes in this PR is currently deployed in one of our environments.

**Comment by faisal-khalique** [comment]
@zanhsieh - can you please check this PR?

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**